astropy/units/function/logarithmic.py

```diff
@@ -277,6 +277,62 @@ def __isub__(self, other):
         self._set_unit(new_unit)
         return self
 
+    def __mul__(self, other):
+        # Multiply by a float or a dimensionless quantity
+        try:
+            # Multiplying a log means putting the factor into the exponent
+            # of the unit
+            new_unit = self.unit._copy(self.unit.physical_unit**other)
+            result = self.view(np.ndarray) * other
+            return self._new_view(result, new_unit)
+        except ValueError:
+            if self.unit.physical_unit == dimensionless_unscaled:
+                # If dimensionless, drop back to normal unit and retry.
+                return self._function_view * other
+            raise UnitTypeError("Cannot multiply logarithmic quantities with "
+                                "anything not dimensionless.")
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __imul__(self, other):
+        try:
+            new_unit = self.unit._copy(self.unit.physical_unit**other)
+            function_view = self._function_view
+            function_view *= other
+            self._set_unit(new_unit)
+            return self
+        except ValueError:
+            raise UnitTypeError("Cannot multiply logarithmic quantities with "
+                                "anything not dimensionless.")
+
+    def __truediv__(self, other):
+        # Divide by a float or a dimensionless quantity
+        try:
+            # Dividing a log means putting the nominator into the exponent
+            # of the unit
+            new_unit = self.unit._copy(self.unit.physical_unit**(1/other))
+            result = self.view(np.ndarray) / other
+            print(result, new_unit)
+            return self._new_view(result, new_unit)
+        except ValueError:
+            if self.unit.physical_unit == dimensionless_unscaled:
+                # If dimensionless, drop back to normal unit and retry.
+                return self._function_view / other
+            raise UnitTypeError("Cannot divide logarithmic quantities by "
+                                "anything not dimensionless.")
+
+    def __itruediv__(self, other):
+        try:
+            new_unit = self.unit._copy(self.unit.physical_unit**(1/other))
```

Comment:
This line makes me think that perhaps one should `except Exception` rather than `except ValueError` since it is possible that `1/other` raises `TypeError`, which then would be in the wrong place.

Reply:
This is also resolved with the `if isinstance(.., Number)`.